avocado/utils/disk.py

```diff
@@ -87,3 +87,28 @@ def get_filesystem_type(mount_point='/'):
             _, fs_file, fs_vfstype, _, _, _ = mount_line.split()
             if fs_file == mount_point:
                 return fs_vfstype
+
+
+def is_linux_fs_type(device):
+    """
+    Checks if specified partition is type 83
+
+    :param device: the device, e.g. /dev/sda3
+
+    :return: False if the supplied partition name is not type 83 linux, True
+            otherwise
+    """
+    disk_device = device.rstrip('0123456789')
+
+    # Parse fdisk output to get partition info.  Ugly but it works.
+    fdisk_fd = os.popen("/sbin/fdisk -l -u '%s'" % disk_device)
```

Comment:
This can be a generic fdisk command, without passing disk name. We can then grep with the device name.
The reason is: For nvme devices, the disk name is /dev/nvme0n1, and for software raid it is /dev/md127.

```
# fdisk -l /dev/nvme0n
fdisk: cannot open /dev/nvme0n: No such file or directory
```

Reply:
But still, the user of the utility function should be responsible for passing the correct *block* device.  @narasimhan-v I don't understand where you get `/dev/nvme0n` from... The block devices in your example are `/dev/nvme0n1` and `/dev/md127`.

Finally, getting all partition information, from all block devices, to be then discarded by grep, is sub-optimal.

Reply:
@clebergnu What I meant was, the `rstrip()` will make nvme0n1 into nvme0n

Reply:
@PraveenPenguin Also, the fdisk output seems to differ between rhel and sles.

SLES:
```
# fdisk -l -u
Disk /dev/nvme0n1: 745.2 GiB, 800166076416 bytes, 195353046 sectors
Units: sectors of 1 * 4096 = 4096 bytes
Sector size (logical/physical): 4096 bytes / 4096 bytes
I/O size (minimum/optimal): 4096 bytes / 4096 bytes
Disklabel type: gpt
Disk identifier: 995E35BB-8574-44FE-9DBF-A956E8D8CEDE

Device          Start       End   Sectors   Size Type
/dev/nvme0n1p1    256      2303      2048     8M PowerPC PReP boot
/dev/nvme0n1p2   2304    526591    524288     2G Linux filesystem
/dev/nvme0n1p3 526592 188431359 187904768 716.8G Linux RAID
```

RHEL:
```
# fdisk -lu

Disk /dev/sda: 59.1 GB, 59055800320 bytes, 14417920 sectors
Units = sectors of 1 * 4096 = 4096 bytes
Sector size (logical/physical): 4096 bytes / 4096 bytes
I/O size (minimum/optimal): 4096 bytes / 4096 bytes
Disk label type: dos
Disk identifier: 0x000c8708

   Device Boot      Start         End      Blocks   Id  System
/dev/sda1   *         256        1279        4096   41  PPC PReP Boot
/dev/sda2            1280      263423     1048576   83  Linux
/dev/sda3          263424    14417919    56617984   8e  Linux LVM

```

Reply:
@narasimhan-v  ok but in `nvme` case it not has `Id ` character present in `fdisk` output , is stand same for RHEL and sles so i need to check how i can adopt    `nvme`  device for using this utility